### PR TITLE
feat: New Module `avm/res/container-registry/registry`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/container-registry/registry`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.container-registry.registry](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.container-registry.registry.yml) |

### CODE REVIEW

File: avm/res/container-registry/registry/main.bicep [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+output loginServer string = reference(registry.id, '2019-05-01').loginServer
+
+@description('The name of the Azure container registry.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The resource ID of the Azure container registry.')
+output resourceId string = registry.id
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(registry.identity, 'principalId') ? registry.identity.principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/container-registry/registry/main.bicep:384]:
```suggestion
output systemAssignedMIPrincipalId string = registry.?identity.?principalId ?? ''
```

File: avm/res/container-registry/registry/main.bicep [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+  @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource. Set to \'\' to disable log collection.')
+  logCategoriesAndGroups: {
+    @description('Optional. Name of a Diagnostic Log category for a resource type this setting is applied to. Set the specific logs to collect here.')
+    category: string?
+
+    @description('Optional. Name of a Diagnostic Log category group for a resource type this setting is applied to. Set to \'AllLogs\' to collect all logs.')
+    categoryGroup: string?
+  }[]?
+
+  @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource. Set to \'\' to disable log collection.')
```
Comment (AlexanderSehr) [inline avm/res/container-registry/registry/main.bicep:513]:
```suggestion
  @description('Optional. The name of metrics that will be streamed. "allMetrics" includes all possible metrics for the resource. Set to \'\' to disable metric collection.')
```

File: avm/res/container-registry/registry/tests/e2e/encr/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    managedIdentities: {
+      userAssignedResourceIds: [
+        nestedDependencies.outputs.managedIdentityResourceId
+      ]
+    }
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/container-registry/registry/tests/e2e/encr/main.test.bicep:76]:
```suggestion
```
Requires an update of the readme

File: avm/res/container-registry/registry/tests/e2e/pe/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      {
+        subnetResourceId: nestedDependencies.outputs.subnetResourceId
+        privateDnsZoneResourceIds: [
+          nestedDependencies.outputs.privateDNSZoneResourceId
+        ]
+        tags: {
+          'hidden-title': 'This is visible in the resource name'
+          Environment: 'Non-Prod'
+          Role: 'DeploymentValidation'
+        }
```
Comment (AlexanderSehr) [inline avm/res/container-registry/registry/tests/e2e/pe/main.test.bicep:65]:
```suggestion
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -32,7 +32,7 @@
 #/avm/res/compute/virtual-machine-scale-set/ @Azure/avm-res-compute-virtualmachinescaleset-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/consumption/budget/ @Azure/avm-res-consumption-budget-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/container-instance/container-group/ @Azure/avm-res-containerinstance-containergroup-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/container-registry/registry/ @Azure/avm-res-containerregistry-registry-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/container-registry/registry/ @Azure/avm-res-containerregistry-registry-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:35]: Shows errors. The team is probably not an approved member of the parent yet. Will resolve once it is.

Comment (JPEasier) [reply .github/CODEOWNERS:35]: Teams are created now

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**